sdks/python/apache_beam/runners/portability/stager.py

```diff
@@ -365,6 +371,11 @@ def create_job_resources(options,  # type: PipelineOptions
             Stager._create_file_stage_to_artifact(
                 pickled_session_file, names.PICKLED_MAIN_SESSION_FILE))
 
+    # stage the submission environment dependencies, if enabled.
```

Comment:
let's provide a way to disable this behavior via experiment. For example:

```
if (log_submission_env_dependencies and not
  options.view_as(DebugOptions).lookup_experiment('disable_logging_submission_environment'))
```

you can use that experiment in one of the tests that you had issues.

Reply:
Done, this is better idea.